### PR TITLE
scheduler: add hot peer filter types

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve? 
Currently, the hot region scheduling does not have a good global view.

### What is changed and how it works?
This PR adds hot peer filter types to let the threshold of hot peer filter be more reasonable.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: server/statistics/hot_cache.go [hunk 1]
```diff
@@ -15,38 +15,133 @@ package statistics
...
+	hotReadRegionMinFlowRate   = 128 * 1024
+)
+
 // HotCache is a cache hold hot regions.
 type HotCache struct {
-	writeFlow *hotPeerCache
-	readFlow  *hotPeerCache
+	writeFlow     *hotPeerCache
+	readFlow      *hotPeerCache
+	hotWriteCache cache.Cache
```
Comment (nolouch) [inline server/statistics/hot_cache.go:39]: I think we can use the old `hotPeerCache` directly.

**Comment by nolouch** [comment]
PTAL @lhy1024

File: server/schedulers/hot_region.go [hunk 2]
```diff
@@ -74,6 +74,12 @@ const (
 
 	minHotScheduleInterval = time.Second
 	maxHotScheduleInterval = 20 * time.Second
+
+	divisor                   = float64(1000) * 2
```
Comment (lhy1024) [inline server/schedulers/hot_region.go:78]: do we need to config `divisor`?

Comment (rleungx) [reply server/schedulers/hot_region.go:78]: We can make it configurable if it is needed in the future.

**Comment by rleungx** [comment]
PTAL @lhy1024 @nolouch

File: server/schedulers/hot_region.go [hunk 3]
```diff
@@ -194,36 +200,67 @@ func (h *hotScheduler) prepareForBalance(cluster opt.Cluster) {
...
+		hotRegionThreshold := getHotRegionThreshold(storesStat, write)
 		h.stLoadInfos[writeLeader] = summaryStoresLoad(
 			storeByte,
 			storeKey,
 			h.pendingSums[writeLeader],
 			regionWrite,
 			minHotDegree,
-			write, core.LeaderKind)
+			hotRegionThreshold,
+			write, core.LeaderKind, mixed)
```
Comment (lhy1024) [inline server/schedulers/hot_region.go:226]: why mixed? It seemed to return false directly.

**Comment by rleungx** [comment]
PTAL @nolouch @lhy1024

File: server/schedulers/hot_region.go (hunk 2 above)
Comment (nolouch) [inline server/schedulers/hot_region.go:78]: how about make it configurable.

File: server/schedulers/hot_region.go [hunk 4]
```diff
@@ -1123,6 +1181,14 @@ func (rw rwType) String() string {
 	}
 }
 
+type hotPeerFilterType int
+
+const (
+	high hotPeerFilterType = iota
```
Comment (nolouch) [inline server/schedulers/hot_region.go:1187]: be a better name?

Comment (rleungx) [reply server/schedulers/hot_region.go:1187]: Any suggestions?

Comment (nolouch) [reply server/schedulers/hot_region.go:1187]: let me change it in the next PR.

**Pull request merged**